### DISCUSSION TITLE
quadrant support on fillSmoothRoundRect

### DISCUSSION THREAD

**Discussion by alex-code** [Bodmer/TFT_eSPI#2333] [New processor/display support or capability request] [answered]
Just stumbled across the latest update with the anti-aliasing arc support, makes such a difference 👍but have a little feature request.

I noticed on `drawSmoothRoundRect` it has quadrant support to make the corners optional.
Would it be possible to add something similar to `fillSmoothRoundRect` so you can have optional square corners?

I'm trying to make a tab style shape (top corners round with bottom square).

**Comment by Bodmer** [comment, marked as answer]
That quadrant support capability is not fully tested yet and some tweaks are needed for corner cases, but setting the inner radius to 1 and the height to 2* radius should work with the existing function. Example:

tft.drawSmoothRoundRect(x, y, radius, 1, w, 2*radius, TFT_WHITE, TFT_BLACK, 0x3);

**Reply by alex-code** [reply]
Thanks for the hint, I was able to make a tab shape using.
```cpp
tft.drawSmoothRoundRect(x, y, r, 1, w - 1, 2 * r, TFT_WHITE, TFT_BLACK, 0x3);
tft.fillRect(x, y + r, w, h - r, TFT_WHITE);
```
